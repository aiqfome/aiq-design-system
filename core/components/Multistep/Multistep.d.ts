import React from 'react';
export interface Props {
    steps: {
        name: string;
        component: any;
        icon?: any;
    }[];
    stepCurrent?: number;
    disabledClickStep?: boolean;
    onClickStep?: (key: any) => void;
}
export interface ButtonProps {
    disabled?: boolean;
}
export declare const Multistep: React.FC<Props>;
//# sourceMappingURL=Multistep.d.ts.map