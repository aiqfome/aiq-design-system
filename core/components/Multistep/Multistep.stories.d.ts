import React from 'react';
declare const _default: {
    component: React.FC<import("./Multistep").Props>;
    title: string;
};
export default _default;
export declare const Basic: React.FC;
export declare const DisabledClickStep: React.FC;
//# sourceMappingURL=Multistep.stories.d.ts.map