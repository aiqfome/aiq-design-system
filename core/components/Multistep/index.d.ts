export * from './Multistep';
//# sourceMappingURL=index.d.ts.map