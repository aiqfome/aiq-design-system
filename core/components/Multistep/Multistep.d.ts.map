{"version":3,"file":"Multistep.d.ts","sourceRoot":"","sources":["../../../src/components/Multistep/Multistep.tsx"],"names":[],"mappings":"AAAA,OAAO,KAA0C,MAAM,OAAO,CAAA;AAU9D,MAAM,WAAW,KAAK;IACpB,KAAK,EAAE;QACL,IAAI,EAAE,MAAM,CAAA;QACZ,SAAS,EAAE,GAAG,CAAA;QACd,IAAI,CAAC,EAAE,GAAG,CAAA;KACX,EAAE,CAAA;IACH,WAAW,CAAC,EAAE,MAAM,CAAA;IACpB,iBAAiB,CAAC,EAAE,OAAO,CAAA;IAC3B,WAAW,CAAC,EAAE,CAAC,GAAG,EAAE,GAAG,KAAK,IAAI,CAAA;CACjC;AAED,MAAM,WAAW,WAAW;IAC1B,QAAQ,CAAC,EAAE,OAAO,CAAA;CACnB;AA2BD,eAAO,MAAM,SAAS,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAoHrC,CAAA"}