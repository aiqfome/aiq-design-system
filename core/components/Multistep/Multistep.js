"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Multistep = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const Text_1 = require("../Text");
const Flex_1 = require("../Flex");
const Box_1 = require("../Box");
const Divider_1 = require("../Divider");
const StepButton = styled_components_1.default.button `
  border: none;
  display: flex;
  background: none;
  align-items: center;
  flex-direction: column;
  min-width: max-content !important;

  cursor: ${({ disabled }) => (disabled ? 'auto' : 'pointer')};
`;
const DividerLeftStyled = styled_components_1.default(Divider_1.Divider) `
  position: absolute;
  top: 12px;
  width: 100%;
  right: calc(50% + 25px);
`;
const DividerRightStyled = styled_components_1.default(Divider_1.Divider) `
  position: absolute;
  top: 12px;
  width: 100%;
  left: calc(50% + 25px);
`;
exports.Multistep = (_a) => {
    var { steps, stepCurrent = 0, disabledClickStep = 0, onClickStep } = _a, props = __rest(_a, ["steps", "stepCurrent", "disabledClickStep", "onClickStep"]);
    const theme = react_1.useContext(styled_components_1.ThemeContext);
    const [stepActive, setStepActive] = react_1.useState(stepCurrent);
    react_1.useEffect(() => {
        if (stepCurrent >= 0 && stepCurrent < steps.length) {
            setStepActive(stepCurrent);
        }
    }, [stepCurrent, steps]);
    function handleClickStep(newStepActive) {
        if (!disabledClickStep) {
            onClickStep && onClickStep(newStepActive);
            setStepActive(newStepActive);
        }
    }
    return (react_1.default.createElement(Flex_1.Flex, Object.assign({ width: '100%', alignItems: 'center', flexDirection: 'column' }, props),
        react_1.default.createElement(Flex_1.Flex, { width: '100%' }, steps.map((step, index) => (react_1.default.createElement(Flex_1.Flex, Object.assign({ flex: 1, mb: '32px', key: index, width: 'auto', alignItems: 'center', flexDirection: 'row', "data-testid": 'steps-step' }, step),
            react_1.default.createElement(Flex_1.Flex, { width: '100%', overflow: 'hidden', position: 'relative', justifyContent: 'center' },
                index > 0 && (react_1.default.createElement(DividerLeftStyled, { color: stepActive >= index
                        ? theme.colors.primary
                        : theme.colors.mediumGrey })),
                react_1.default.createElement(StepButton, Object.assign({}, step, { type: 'button', "data-testid": 'step-button', disabled: !!disabledClickStep, onClick: () => handleClickStep(index) }),
                    step.icon || (react_1.default.createElement(Flex_1.Flex, { backgroundColor: stepActive >= index
                            ? theme.colors.primary
                            : theme.colors.mediumGrey, height: '24px', width: '24px', justifyContent: 'center', alignItems: 'center', marginBottom: '12px', borderRadius: '50px' },
                        react_1.default.createElement(Text_1.Text, { color: 'white', cursor: disabledClickStep ? 'auto' : 'pointer' }, index + 1))),
                    react_1.default.createElement(Text_1.Text, { fontSize: 'medium', fontWeight: 'medium', cursor: disabledClickStep ? 'auto' : 'pointer', color: stepActive >= index
                            ? theme.colors.primary
                            : theme.colors.mediumGrey }, step.name)),
                steps.length - 1 > index && (react_1.default.createElement(DividerRightStyled, { color: stepActive > index
                        ? theme.colors.primary
                        : theme.colors.mediumGrey }))))))),
        steps.map((step, index) => stepActive === index && react_1.default.createElement(Box_1.Box, { key: 'index' }, step.component))));
};
exports.Multistep.propTypes = {
    steps: prop_types_1.default.array.isRequired,
    stepCurrent: prop_types_1.default.number,
    disabledClickStep: prop_types_1.default.bool,
    onClickStep: prop_types_1.default.func
};
