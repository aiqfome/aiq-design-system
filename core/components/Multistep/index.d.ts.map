{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/Multistep/index.tsx"],"names":[],"mappings":"AAAA,cAAc,aAAa,CAAA"}