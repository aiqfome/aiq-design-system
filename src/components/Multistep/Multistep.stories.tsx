import React, { useState } from 'react'

import { Flex } from '../Flex'
import { Button } from '../Button'

import { Multistep } from './Multistep'

export default {
  component: Multistep,
  title: 'Multistep'
}

// eslint-disable-next-line react/prop-types
const Step: React.FC = ({ children }) => {
  return <h1>{children}</h1>
}

const steps = [
  { name: 'StepOne', component: <Step>One</Step> },
  { name: 'StepTwo', component: <Step>Two</Step> },
  { name: 'StepThree', component: <Step>Three</Step> },
  { name: 'StepFour', component: <Step>Four</Step> }
]

export const Basic: React.FC = () => {
  const [stepCurrent, setStepCurrent] = useState(0)

  function handleClickBtnNext() {
    setStepCurrent(stepCurrent + 1)
  }

  function handleClickBtnLast() {
    setStepCurrent(stepCurrent - 1)
  }

  return (
    <Flex flexDirection='column'>
      <Multistep stepCurrent={stepCurrent} steps={steps} />
      <Flex flexDirection='row' marginTop='16px' variant='centralized'>
        <Button
          onClick={handleClickBtnLast}
          variant='contained'
          palette='primary'
          marginRight={'16px'}
        >
          Last
        </Button>
        <Button
          onClick={handleClickBtnNext}
          variant='contained'
          palette='primary'
        >
          Next
        </Button>
      </Flex>
    </Flex>
  )
}

export const DisabledClickStep: React.FC = () => {
  const [stepCurrent, setStepCurrent] = useState(0)

  function handleClickBtnNext() {
    setStepCurrent(stepCurrent + 1)
  }

  function handleClickBtnLast() {
    setStepCurrent(stepCurrent - 1)
  }

  return (
    <Flex flexDirection='column'>
      <Multistep
        disabledClickStep={true}
        stepCurrent={stepCurrent}
        steps={steps}
      />
      <Flex flexDirection='row' marginTop='16px' variant='centralized'>
        <Button
          onClick={handleClickBtnLast}
          variant='contained'
          palette='primary'
          marginRight={'16px'}
        >
          Last
        </Button>
        <Button
          onClick={handleClickBtnNext}
          variant='contained'
          palette='primary'
        >
          Next
        </Button>
      </Flex>
    </Flex>
  )
}
