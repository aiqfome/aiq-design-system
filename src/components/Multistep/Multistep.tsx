import React, { useState, useContext, useEffect } from 'react'
import PropTypes from 'prop-types'

import styled, { ThemeContext } from 'styled-components'

import { Text } from '../Text'
import { Flex } from '../Flex'
import { Box } from '../Box'
import { Divider } from '../Divider'

export interface Props {
  steps: {
    name: string
    component: any
  }[]
  stepCurrent?: number
  disabledClickStep?: boolean
  onClickStep?: () => void
}

const StepButton = styled.button`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 28px;
  background: none;
  border: none;
`

export const Multistep: React.FC<Props> = ({
  steps,
  stepCurrent = 0,
  disabledClickStep = 0,
  onClickStep,
  ...props
}) => {
  const theme = useContext(ThemeContext)
  const [stepActive, setStepActive] = useState(stepCurrent)

  useEffect(() => {
    if (stepCurrent >= 0 && stepCurrent < steps.length) {
      setStepActive(stepCurrent)
    }
  }, [stepCurrent, steps])

  function handleClickStep(newStepActive) {
    onClickStep && onClickStep()
    if (!disabledClickStep) {
      setStepActive(newStepActive)
    }
  }

  return (
    <Flex width='100%' alignItems='center' flexDirection='column' {...props}>
      <Flex width='100%'>
        {steps.map((step, index) => (
          <Flex
            key={index}
            marginBottom='32px'
            flexDirection='row'
            alignItems='center'
            width={index === steps.length - 1 ? '80px' : '100%'}
          >
            <StepButton type='button' onClick={() => handleClickStep(index)}>
              <Flex
                backgroundColor={
                  stepActive >= index
                    ? theme.colors.primary
                    : theme.colors.mediumGrey
                }
                height='24px'
                width='24px'
                justifyContent='center'
                alignItems='center'
                marginBottom='12px'
                borderRadius='50px'
              >
                <Text cursor='pointer' color='white'>
                  {index + 1}
                </Text>
              </Flex>
              <Text
                fontWeight='medium'
                fontSize='medium'
                cursor='pointer'
                color={
                  stepActive >= index
                    ? theme.colors.primary
                    : theme.colors.mediumGrey
                }
              >
                {step.name}
              </Text>
            </StepButton>

            {steps.length - 1 > index && (
              <Divider
                color={
                  stepActive > index
                    ? theme.colors.primary
                    : theme.colors.mediumGrey
                }
                width='100%'
              />
            )}
          </Flex>
        ))}
      </Flex>

      {steps.map(
        (step, index) =>
          stepActive === index && <Box key='index'>{step.component}</Box>
      )}
    </Flex>
  )
}

Multistep.propTypes = {
  steps: PropTypes.array.isRequired,
  stepCurrent: PropTypes.number,
  disabledClickStep: PropTypes.bool,
  onClickStep: PropTypes.func
}
