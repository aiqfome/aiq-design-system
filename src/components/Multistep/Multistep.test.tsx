import React from 'react'

import { render } from '../utils/test/render'

import { Multistep } from './Multistep'

// eslint-disable-next-line react/prop-types
const Step: React.FC = ({ children }) => {
  return <h1>{children}</h1>
}

const steps = [
  { name: 'StepOne', component: <Step>One</Step> },
  { name: 'StepTwo', component: <Step>Two</Step> },
  { name: 'StepThree', component: <Step>Three</Step> },
  { name: 'StepFour', component: <Step>Four</Step> }
]

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(
      <Multistep disabledClickStep={true} stepCurrent={0} steps={steps} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
