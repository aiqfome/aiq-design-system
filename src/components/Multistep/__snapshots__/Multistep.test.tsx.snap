// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-gtssRu edRxNW"
  width="100%"
>
  <div
    className="sc-gtssRu fCrJhm"
    width="100%"
  >
    <div
      className="sc-gtssRu jNNDkS"
      width="100%"
    >
      <button
        className="sc-eCApGN hjZHXr"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-gtssRu dfpfjj"
          height="24px"
          width="24px"
        >
          <span
            className="sc-bdnylx fNIHTr"
            color="white"
            cursor="pointer"
          >
            1
          </span>
        </div>
        <span
          className="sc-bdnylx hMBpez"
          color="#eb2f96"
          cursor="pointer"
          fontSize="medium"
          fontWeight="medium"
        >
          StepOne
        </span>
      </button>
      <hr
        className="sc-hKFyIo yhYLO"
        height="1px"
        width="100%"
      />
    </div>
    <div
      className="sc-gtssRu jNNDkS"
      width="100%"
    >
      <button
        className="sc-eCApGN hjZHXr"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-gtssRu fzuEiW"
          height="24px"
          width="24px"
        >
          <span
            className="sc-bdnylx fNIHTr"
            color="white"
            cursor="pointer"
          >
            2
          </span>
        </div>
        <span
          className="sc-bdnylx kkZeom"
          color="#dedede"
          cursor="pointer"
          fontSize="medium"
          fontWeight="medium"
        >
          StepTwo
        </span>
      </button>
      <hr
        className="sc-hKFyIo yhYLO"
        height="1px"
        width="100%"
      />
    </div>
    <div
      className="sc-gtssRu jNNDkS"
      width="100%"
    >
      <button
        className="sc-eCApGN hjZHXr"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-gtssRu fzuEiW"
          height="24px"
          width="24px"
        >
          <span
            className="sc-bdnylx fNIHTr"
            color="white"
            cursor="pointer"
          >
            3
          </span>
        </div>
        <span
          className="sc-bdnylx kkZeom"
          color="#dedede"
          cursor="pointer"
          fontSize="medium"
          fontWeight="medium"
        >
          StepThree
        </span>
      </button>
      <hr
        className="sc-hKFyIo yhYLO"
        height="1px"
        width="100%"
      />
    </div>
    <div
      className="sc-gtssRu drexdK"
      width="80px"
    >
      <button
        className="sc-eCApGN hjZHXr"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-gtssRu fzuEiW"
          height="24px"
          width="24px"
        >
          <span
            className="sc-bdnylx fNIHTr"
            color="white"
            cursor="pointer"
          >
            4
          </span>
        </div>
        <span
          className="sc-bdnylx kkZeom"
          color="#dedede"
          cursor="pointer"
          fontSize="medium"
          fontWeight="medium"
        >
          StepFour
        </span>
      </button>
    </div>
  </div>
  <div
    className="sc-dlnjPT ldnTBd"
  >
    <h1>
      One
    </h1>
  </div>
</div>
`;
